spec/controllers/api/profile_controller_spec.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe Api::ProfileController do
+  include ControllerHelpers
+
+  let(:bob) { users(:bob) }
+
+  context 'PATCH update' do
+    it 'updates bobs preferred_locale to german' do
+      login_as(:bob)
+
+      preferred_locale_params = {
+        data: {
+          attributes: {
+            preferred_locale: 'de'
+          }
+        }
+      }
+
+      patch :update, params: preferred_locale_params
+
+      expect(bob.preferred_locale).to eq('de')
```

Comment:
bob.reload.pref .... you need to reload entry bob. 
what was his locale before? make sure it's changed after this call